### ISSUE TITLE
 `Argument of type '() => Promise<void>' is not assignable to parameter of type 'PrismaPromise<any>[]'.` when using `prisma.$transaction` after extending client

### ISSUE THREAD

**Issue by tuarrep** [47ng/prisma-field-encryption#116]
The client is initialized like this:
```js
const client = new PrismaClient({
    log: ["query"],
  })
.$extends(fieldEncryptionExtension());
```

Next, we use interactive transactions :
```js
await prisma.$transaction(async () => {
    await prisma.beneficiary.update({
      where},
      data,
    });

    await prisma.invitation.update({
      where,
      data,
    });
  });
  ```

No problem before extending client with `fieldEncryptionExtension()` and this error after:
```bash
- error TS2345: Argument of type '() => Promise<void>' is not assignable to parameter of type 'PrismaPromise<any>[]'.

98   await prisma.$transaction(async () => {
                               ~~~~~~~~~~~~~

    at createTSError (/Users/nicolas/Dev/xxx/node_modules/ts-node/src/index.ts:859:12)
    at reportTSError (/Users/nicolas/Dev/xxx/node_modules/ts-node/src/index.ts:863:19)
    at getOutput (/Users/nicolas/Dev/xxx/node_modules/ts-node/src/index.ts:1077:36)
    at Object.compile (/Users/nicolas/Dev/xxx/node_modules/ts-node/src/index.ts:1433:41)
    at Module.m._compile (/Users/nicolas/Dev/xxx/node_modules/ts-node/src/index.ts:1617:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1551:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/nicolas/Dev/xxx/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1282:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1098:12)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14) {
  diagnosticCodes: [ 2345 ]
}

```

Versions:
- `prisma` and `@prisma/client`: 5.17.0
- `prisma-field-encryption`: 1.5.2

**Comment by franky47** [comment]
What happens if you use the first argument of your transaction callback as the Prisma client? This is the way I've always seen it done in the docs and examples:

```ts
await prisma.$transaction(async (tx) => {
    await tx.beneficiary.update({
      where},
      data,
    });

    await tx.invitation.update({
      where,
      data,
    });
  });
```

**Comment by tuarrep** [comment]
Thanks for pointing this out.

Unfortunately, it causes the same Typescript error.

Even when trying to type `tx` as `PrismaClient`

**Comment by franky47** [comment]
I just realised I had had this issue before, and opened one upstream on Prisma: https://github.com/prisma/prisma/issues/20326.

In the mean time, I fixed it internally by casting the extended client type back to the original one. Since `prisma-field-encryption` acts transparently with regards to queries and mutations, the type annotation is safe just for this extension (ie: if you do an extension chain, you might want to tweak the type you are casting to, or use an intermediate variable).

https://github.com/47ng/prisma-field-encryption/blob/8e4769abbb015bb224623bbde09361f822adbeea/src/tests/prismaClient.ts#L21
